prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt

```diff
@@ -0,0 +1,154 @@
+package gov.cdc.prime.router.fhirengine.translation
+
+import ca.uhn.hl7v2.DefaultHapiContext
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer
+import gov.cdc.prime.router.fhirengine.utils.FhirTranscoder
+import org.apache.logging.log4j.kotlin.Logging
+
+/**
+ * Instance of this manager can be used to validate and update translation schemas in various environments
+ */
+class TranslationSchemaManager : Logging {
+
+    /**
+     * This enum contains the kinds of schemas that currently can be managed
+     */
+    enum class SchemaType(val directory: String, val outputExtension: String) {
+        FHIR("fhir_transforms", "fhir"),
+        HL7("hl7_mapping", "hl7"),
+    }
+
+    companion object {
+
+        private val context = DefaultHapiContext()
+        private val hl7Parser = context.getGenericParser()
+
+        data class ValidationResults(
+            val path: String,
+            val passes: Boolean,
+            val didError: Boolean = false,
+        )
+
+        data class ValidationContainer(val input: String, val output: String, val schemaUri: String)
+
+        class TranslationValidationException(override val message: String, override val cause: Throwable? = null) :
+            RuntimeException(cause)
+
+        /**
+         * Internal helper function that fetches the raw input, output file and the URI for the schema to be validated
+         *
+         * @property blobs the list of blobs to search through to find the schema
+         * @property blobContainerInfo the information on the blob container to download from
+         * @property inputDirectoryPath the path to the directory currently being processed
+         * @property schemaType the kind of transform being validated
+         * @return [ValidationContainer]
+         * @throws [TranslationValidationException] if an error occurred gathering the input/output or finding the schema
+         */
+        internal fun getRawInputOutputAndSchemaUri(
+            blobs: List<BlobAccess.Companion.BlobItemAndPreviousVersions>,
+            blobContainerInfo: BlobAccess.BlobContainerMetadata,
+            inputDirectoryPath: String,
+            schemaType: SchemaType,
+        ): ValidationContainer {
+            val inputBlobUrl = "${blobContainerInfo.getBlobEndpoint()}/${inputDirectoryPath}input.fhir"
```

Comment:
would something like `input.*` be more flexible going forward?

Reply:
Yea, I went back and forth on that.  I'm happy to switch it over